### PR TITLE
Add Hook to add custom criteria for product sliders

### PR DESCRIPTION
**Pull request by egoist-dd**
## Description

Please describe your pull request:
- add custom filters to modify slider data, e.g. exclude articles or manufacturers

| Questions | Answers |
| --- | --- |
| BC breaks? | no |
| Tests pass? | yes |
| Related tickets? | no |
| How to test? | add new hook for Shopware_Controllers_Widgets_Emotion_GetProductSliderData and add own or custom filters |

### CODE REVIEW

File: engine/Shopware/Controllers/Widgets/Emotion.php [hunk 1]
```diff
@@ -773,6 +773,12 @@ private function getProductSliderData($category, $offset = 0, $limit, $sort = nu
         $factory = Shopware()->Container()->get('shopware_search.store_front_criteria_factory');
         $criteria = $factory->createBaseCriteria([$category], $context);
 
+        $criteria = Shopware()->Events()->filter(
```
Comment (bcremer) [inline engine/Shopware/Controllers/Widgets/Emotion.php]: Would't it make sense to move this event down after the switch statement just before the call to `search()`  so the completed criteria objects can be modified?

**Comment by egoist-dd** [comment]
of course, i moved it down :-)

**Comment by janbuecker** [comment]
Hey @egoist-dd, thanks for your contribution.

This method is no longer available in Shopware 5.3. In Shopware 5.3 we've added proper extension points for shopping worlds.

**Comment by egoist-dd** [comment]
@janbuecker 

i take a look in your new components, but there is no chance to add own criterias. The method generateCriteria is still private and has no hookpoint.

Can you tell me how to add an critieria for modifying my search?

**Comment by janbuecker** [comment]
Every component now has a component handler which is also defined in the DI container. For your use-case, you can simply decorate the `shopware_emotion.component_handler.article_slider` service and bring in your own logic.

To learn more about the changes, please have a look at the `UPGRADE-5.3.md`.

**Comment by egoist-dd** [comment]
Hi Jan, we are currently migrate to shopware 5.3.

I take a look in the new component but if i'd like to add a single article slider to our view the new component is not used.

Is there any way to re open the pull request to add custom criterias?

**Pull request closed without merging**